File: src/main/java/com/rdvdev2/timetravelmod/api/event/WorldCorruptionChangedEvent.java
package com.rdvdev2.timetravelmod.api.event;

import com.rdvdev2.timetravelmod.api.dimension.ITimeline;
import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;

public interface WorldCorruptionChangedEvent {

    Event<WorldCorruptionChangedEvent> EVENT = EventFactory.createArrayBacked(WorldCorruptionChangedEvent.class,
            listeners -> (timeline, value, previousValue) -> {
                for (WorldCorruptionChangedEvent listener: listeners) {
                    listener.onWorldCorruptionChanged(timeline, value, previousValue);
                }
            });

    void onWorldCorruptionChanged(ITimeline timeline, int value, int previousValue);
}

File: src/main/kotlin/com/rdvdev2/timetravelmod/api/event/WorldCorruptionChangedEvent.kt
package com.rdvdev2.timetravelmod.api.event

import com.rdvdev2.timetravelmod.api.dimension.ITimeline
import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory

interface WorldCorruptionChangedEvent {
    fun onWorldCorruptionChanged(timeline: ITimeline, value: Int, previousValue: Int)

    companion object {
        @JvmField
        val EVENT: Event<WorldCorruptionChangedEvent> = EventFactory.createArrayBacked(WorldCorruptionChangedEvent::class.java) { listeners ->
            object : WorldCorruptionChangedEvent {
                override fun onWorldCorruptionChanged(timeline: ITimeline, value: Int, previousValue: Int) {
                    for (listener in listeners) {
                        listener.onWorldCorruptionChanged(timeline, value, previousValue)
                    }
                }
            }
        }
    }
}